doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+# Implementing a face beautification algorithm on G-API {#tutorial_gapi_face_beautification}
+
+@prev_tutorial{tutorial_gapi_anisotropic_segmentation}
+
+[TOC]
+
+# Introduction {#gapi_fb_intro}
+
+In this tutorial you will learn:
+* How to infer different networks inside the pipeline with G-API;
+* How to process a videostream inside the pipeline with G-API;
+* The main ideas of the Face Beautification algorithm.
+
+You can find the source code in
+`samples/cpp/tutorial_code/face_beautification.cpp`
+of the OpenCV library source tree.
+
+It is a well-known fact that geterogeneous algorithms based on Computer Vision
+image processing with the backup of deep neural networks inference
+(Deep Learning) are able to solve a huge variety of problems which CV-only
+ones might experience difficulties with. Let's consider the certain
+algorithm as an example.
+
+# Face Beautification algorithm {#gapi_fb_algorithm}
+
+The algorithm of retouching a face has been chosen as the subject for this
+tutorial. Its main idea is to implement a real-time
+videostream processing that detects faces and applies some filters to make them
+look beautiful (more or less). The pipeline is the following:
+
+![The Face Beautification algorithm](pics/algo.png)
+
+The algorithm consists of two parts: an inference of two networks (including
+data pre- and post-processing) and an image filtering pipeline which uses the
+inference data to create masks, then applies filters and composes the output
+retouched image. What is each of these parts for? To be more certain, the
+"beautification" implemented here is smoothing out the skin (by the
+`Bilateral filter`) and making eyes, a nose and a mouth sharper (by the
+`Unsharp mask`). Thus, to know which filter should be applied to which area of
+image, we need to calculate a contour of the face and contours of facial
+elements --- we can get them from data received via deep learning methods.
+
+Two topologies from OMZ have been used in this sample: the
+<a href="https://github.com/opencv/open_model_zoo/tree/master/models/intel
+/face-detection-adas-0001">face-detection-adas-0001</a>
+and the
+<a href="https://github.com/opencv/open_model_zoo/blob/master/models/intel
+/facial-landmarks-35-adas-0002/description/facial-landmarks-35-adas-0002.md">
+facial-landmarks-35-adas-0002</a>.
+
+The face detector takes the input image and returns a blob with the shape
+[1, 1, N, 7] after the inference, where `N = 200` is the maximum number of
+faces which can be detected. Obviously, the output `cv::Mat` post-processing is
```

Comment:
Not that obviously.

"In order to process every face individually, we need to convert this result to a list of regions in the image". And I'd omit all subsequent details, those must be clear from the code itself